### DISCUSSION TITLE
update/slide-out: sync with remote parent branch and rebase?

### DISCUSSION THREAD

**Discussion by levinotik** [VirtusLab/git-machete#1084] [Q&A] [answered]
Consider the following status:

```
    └─main
      │
      └─levi_notik/add-healthcheck-to-celery (untracked)
```

Essentially, I've branched from `main` a while ago and have not since fetched the latest from the remote (`origin/main`). My typical workflow prior to opening a PR would be to fetch the latest of `origin/main` and then squash and rebase my branch on top of it so I can get a clean PR with no conflicts and a simple fast-forward (if I wanted to do a FF). 

I saw that mach has a `slide-out` which sounds like it's the inverse of what I want, i.e. if I had a bunch of child branches or I had `main` checked out, that'd be the way to get all the children synced.

I also saw `git mach update` which I think _is_ exactly what I'm looking for. Help says it will

> Sync the current branch with its upstream (parent) branch via rebase or merge

Does this mean it'll do a fetch of the latest from upstream and then ask me whether I want to rebase or merge? In general, is `update`, in fact, what I should be using to accomplish this?

Thanks in advance for the help!

**Comment by PawelLipski** [comment, marked as answer]
Hey, you can use `git machete traverse --fetch` which will first fetch all remotes and then walk through you branches and "get them sorted" one by one. In your case, it's gonna fast-forward `master` to `origin/master`, then rebase your branch onto `master`, and then (force-)push your branch.

`git machete update` is an older & narrower command, which basically just does the rebase of the current branch onto its parent ✂️

**Reply by levinotik** [reply]
Incredible. What _can't_ machete do?? Thanks a lot. I'll take this for a spin. Is there a way to do all of that but skip the final operation, i.e. the (force-push)?

**Reply by PawelLipski** [reply]
Haha thanks 😅 

>  but skip the final operation, i.e. the (force-push)?

There are three ways to make `git machete traverse` skip push (either for a specific branch, or at all): either:

1. use _qualifiers_ for a specific branch:

```
main
    levi_notik/add-healthcheck-to-celery  push=no
```

2. or pass `--no-push`/`--no-push-untracked` to `git machete traverse`
3. or set `git config machete.traverse.push false`

See https://git-machete.readthedocs.io/#traverse for more details ;)

**Reply by PawelLipski** [reply]
Also, pls check [git-machete-intellij-plugin](https://github.com/VirtusLab/git-machete-intellij-plugin) if you're using IntelliJ
